File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        jump = 0
        pos = 0

        while pos < len(nums)-1:
            if pos + nums[pos] >= len(nums) - 1:
                return jump + 1

            start = pos
            for i in range(1, nums[start]+1):
                if nums[start + i] + start + i > nums[pos] + pos:
                    pos = start + i
            jump += 1
        
        return jump






        

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
            
        int curFarthest = nums[0];
        int curEnd = 0;
        int result = 0;

        for (int i = 0; i < nums.size() - 1; ++i) {
            curFarthest = max(curFarthest, i + nums[i]);
            if (i == curEnd) {
                curEnd = curFarthest;
                result++;

                if (curEnd >= nums.size() - 1) {
                    break;
                }
            }
        }
        return result;
    }
};